File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/Services/NavigationService/DetailViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO

Namespace MVVMDemo.Services.Navigation

    Public Class DetailViewModel
        Implements ISupportParameter

        Public Overridable Property Person As PersonInfo

        Private _Parameter As Object

        Private Property Parameter As Object Implements ISupportParameter.Parameter
            Get
                Return _Parameter
            End Get

            Set(ByVal value As Object)
                _Parameter = value
                Person = CType(_Parameter, PersonInfo)
            End Set
        End Property

        Private ReadOnly Property NavigationService As INavigationService
            Get
                Return GetRequiredService(Of INavigationService)()
            End Get
        End Property

        Public Sub Back()
            NavigationService.GoBack()
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/Services/NavigationService/DetailViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;

namespace MVVMDemo.Services.Navigation {
    public class DetailViewModel : ISupportParameter {
        public virtual PersonInfo Person { get; set; }

        object _Parameter;
        object ISupportParameter.Parameter {
            get { return _Parameter; }
            set {
                _Parameter = value;
                Person = (PersonInfo)_Parameter;
            }
        }

        INavigationService NavigationService { get { return this.GetRequiredService<INavigationService>(); } }
        public void Back() {
            NavigationService.GoBack();
        }
    }
}
